packages/zapp/console/src/components/flytegraph/ReactFlow/transformDAGToReactFlowV2.tsx

```diff
@@ -61,13 +61,13 @@ export const buildReactFlowDataProps = (props: BuildDataProps) => {
     currentNestedView,
   } = props;
 
-  const taskType = node.value?.template ? node.value.template.type : null;
-  const displayName = node.name;
+  const { value: nodeValue, name: displayName, scopedId, type: nodeType } = node || {};
+  const taskType = nodeValue?.template?.type ?? null;
```

Comment:
❤️ this destructure, but do we really need `|| {}` as `node` is required in props?

Reply:
@ursucarina 
nit: Some info I talked about at standup: 
- buildReactFlowDataProps does not need export keyword, as used only in this file
- buildReactFlowNode doesn't need export keyword
- In buildGraphMapping we need to add filtering prior to map in :
``` contextNode.nodes.map((node: dNode) => {``` 
this will guarantee that any node coming as input to  buildReactFlowDataProps would exist, so `node || {}` check won't be needed, and similar to other `?` checks Olga pointed